modules/memfault-firmware-sdk/memfault_integration.c

```diff
@@ -148,13 +148,13 @@ static int init(const struct device *unused)
 		}
 	}
 
-#ifdef CONFIG_MEMFAULT_NCS_DEVICE_ID_IMEI
-	err = device_info_init();
-	if (err) {
-		LOG_ERR("Device info initialization failed, error: %d", err);
+	if (!IS_ENABLED(CONFIG_MEMFAULT_DEVICE_INFO_CUSTOM) &&
+	    IS_ENABLED(CONFIG_MEMFAULT_NCS_DEVICE_ID_IMEI)) {
+		err = device_info_init();
```

Comment:
Shouldn't device_info_init() be known at compilation time?
Call will be removed by the compiler, not the linker.

If my comment dosen't make sense, please disregard it.

Reply:
You are right. This does cause a compilation warning. I updated.